Testing/UnitTests/xASL_ut_UnitTest_function_StdNan.m

```diff
@@ -0,0 +1,66 @@
+function UnitTest = xASL_ut_UnitTest_function_StdNan(TestRepository)
+%xASL_ut_UnitTest_function_CheckFileCount Individual unit test for xASL_adm_CheckFileCount
+%
+% INPUT:        TestRepository - Path to test repository.
+%
+% OUTPUT:       UnitTest  - Test structure
+%               name      - Name of tested module or submodule (char array)
+%               unit      - Insert one of the following: 'Module', 'Submodule' or 'Function'
+%               passed    - Result of all subtests combined (true or false)
+%               test      - Structure with individual subtest results
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:  Should be run using xASL_ut_UnitTesting.
+%
+% EXAMPLE:      UnitTests(1) = xASL_ut_UnitTest_function_CheckFileCount(TestRepository);
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% Copyright 2015-2021 ExploreASL
+
+%% Initialize test structure
+
+% Insert test name here
+UnitTest.name = 'xASL_stat_StdNan';
+
+% Define whether you are testing a module, submodule or function
+UnitTest.unit = 'Function';
+
+%% Test run 1
+
+% Give your individual subtest a name
+UnitTest.tests(1).testname = 'Multiple test calls with different arrays';
+
+% Start the test
+testTime = tic;
+
+% Run your test here
+y1 = xASL_stat_StdNan([0, 0, 0, 0, 0]);
+y2 = xASL_stat_StdNan([0, 1, 2, 3, 4]);
```

Comment:
@MichaelStritt 
I know that we are not ready to sweep across a number of parameters, but this would be the desired unit test, here we only test a limited amount of values. What do you think of creating a function `xASL_ut_CreateInputValues`, where you provide the upper and lower limit as well as the "resolution/number bins" of the values to test. So e.g., upper limit of 1,000,000 and lower limit of -1,000,000 with resolution 10 would create as input parameters for the unit test:
>> [-1000000:2000000/10:1000000]

ans =

  Columns 1 through 3

    -1000000     -800000     -600000

  Columns 4 through 6

     -400000     -200000           0

  Columns 7 through 9

      200000      400000      600000

  Columns 10 through 11

      800000     1000000

Also, we now do this unit test by assuming that another unit (==Matlab's `std`) is perfect. Whereas, this unit `xASL_stat_StdNaN` probably uses Matlab's `std` internally :) And whereas Matlab's std may differ between versions (theoretically)

Reply:
> Also, we now do this unit test by assuming that another unit (==Matlab's std) is perfect

I don't understand what you would do differently. Do you want to open an old math book and use reference values? 

> And whereas Matlab's std may differ between versions (theoretically)

Same here. Matlab is tested and approved software. If you don't trust Matlab to this level, you shouldn't use it at all.

Reply:
> I know that we are not ready to sweep across a number of parameters, but this would be the desired unit test, here we only test a limited amount of values. 

Check out my comments for `xASL_ut_UnitTest_function_str2num`. We can implement this, but I don't think it's needed right now. Let's start as simple as possible. We should focus on testing more scripts and not on testing the few ones more thoroughly.